src/indicators/drought/chirpsgefs_rainfallforecast.py

```diff
@@ -0,0 +1,227 @@
+
+    for thres in ds_thresh_list:
+        # compute the percentage of the admin area that has cells below the threshold
+        # set all values with below average rainfall to 1 and others to 0
+        forecast_binary = np.where(ds.values <= thres, 1, 0)
+        # compute number of cells in admin region (sum) and number of cells in admin region with below average rainfall (count)
+        bin_zonal = pd.DataFrame(
+            zonal_stats(
+                vectors=df,
+                raster=forecast_binary,
+                affine=raster_transform,
+                stats=["count", "sum"],
+                nodata=np.nan,
+            )
+        )
+        df[f"perc_se{thres}"] = bin_zonal["sum"] / bin_zonal["count"] * 100
+
+    return df
+
+
+def compute_stats_rainyseason(
+    country_name,
+    config,
+    adm_level,
+    days_ahead,
+    output_dir,
+    rainy_season_path,
+    use_cache=True,
+):
+    # this takes some time to compute, couple of hours at max
+    parameters = config.parameters(country_name)
+    country_iso3 = parameters["iso3_code"].lower()
+    adm_col = parameters[f"shp_adm{adm_level}c"]
+    # TODO: find a more dynamic method to define this path
+    adm_bound_path = (
+        Path(config.DATA_DIR)
+        / config.PUBLIC_DIR
+        / config.RAW_DIR
+        / country_iso3
+        / config.SHAPEFILE_DIR
+        / parameters[f"path_admin{adm_level}_shp"]
+    )
+    rainy_dates = get_rainy_season_dates(rainy_season_path)
+    gdf_adm = gpd.read_file(adm_bound_path)
+
+    # load the tif file for each date and compute the statistics
+    for d in rainy_dates:
+        date_str = pd.to_datetime(d).strftime("%Y%m%d")
+        output_path = (
+            output_dir
+            / f"{country_iso3}_chirpsgefs_stats_adm{adm_level}_{days_ahead}days_{date_str}.csv"
+        )
+
+        if use_cache and output_path.exists():
+            logger.debug(
+                f"{output_path} already exists and cache is set to True, skipping"
+            )
+        else:
+            Path(output_path.parent).mkdir(parents=True, exist_ok=True)
+            chirpsgefs_raster_filepath = (
+                config.CHIRPSGEFS_RAW_DIR
+                / config.CHIRPSGEFS_RAW_FILENAME.format(
+                    days_ahead=days_ahead, date=date_str
+                )
+            )
+            if not chirpsgefs_raster_filepath.exists():
+                logger.info(
+                    f"CHIRPS-GEFS data for {date_str} was not found, {chirpsgefs_raster_filepath}"
+                )
+            else:
+                logger.debug(f"Retrieving stats for {date_str}")
+                rds = rioxarray.open_rasterio(chirpsgefs_raster_filepath)
+                df = compute_raster_stats(
+                    rds.sel(band=1), rds.rio.transform(), gdf_adm, adm_col
+                )
+                df.loc[:, "date"] = d
+                df.loc[:, "date_forec_end"] = df.loc[:, "date"] + timedelta(
+                    days=days_ahead - 1
```

Comment:
Why -1 here?

Reply:
cause the `days_ahead` is the number of days the forecast is valid for. The first date of the forecast is extracted from the filename, so date `days_ahead-1` ahead - current date = days_ahead